src/containers/simdset.d

```diff
@@ -0,0 +1,226 @@
+/**
+ * SIMD-accelerated Set
+ * Copyright: © 2015 Economic Modeling Specialists, Intl.
+ * Authors: Brian Schott
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt Boost License 1.0)
+ */
+module containers.simdset;
+
+/**
+ * Set implementation that is well suited for small sets and simple items.
+ *
+ * Uses SSE instructions to compare multiple elements simultaneously. SimdSet
+ * has the following time complexities for contains:
+ * $(UL
+ * $(LI T == ubyte: n / 16)
+ * $(LI T == ushort: n / 8)
+ * $(LI T == uint: n / 4)
+ * $(LI T == ulong: n / 2)
+ * )
+ *
+ * Note: Only works on x86_64. Does NOT add GC ranges. Do not store pointers in
+ * this container unless they are also stored somewhere else.
+ */
+version (D_InlineAsm_X86_64) struct SimdSet(T)
```

Comment:
You should put a template constraint on `T` to make sure that bad things don't happen for non-integer types.